client/src/cluster.rs

```diff
@@ -116,7 +118,7 @@ mod tests {
         let url = "http://my-url.com:7000/";
```

Comment:
Can we add an additional test using TLS. Given  `https://my-url.com:7000/"`, then `"wss://my-url.com:7001/"` should be used.

Reply:
Yeah, will do that real quick.

Reply:
Added https testing for both port and no_port, just like the http tests, and added an if block to change to wss or ws based on what scheme url is already using.